pkg/components: add Inspektor Gadget component

iaguis opened the pull request:
Fixes #1063

invidian commented on assets/charts/components/inspektor-gadget/templates/psp.yaml:
```diff
@@ -0,0 +1,24 @@
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: inspektor-gadget-restricted
```
`-restricted`? From what? :joy: This is almost the most permissive policy I've seen :smile: 
```suggestion
  name: inspektor-gadget
```

iaguis replied:
at least it can't use Secrets? :joy:

invidian commented on assets/charts/components/inspektor-gadget/values.yaml:
```diff
@@ -0,0 +1,42 @@
+# Declare variables to be passed into your templates.
+
+image:
+  repository: docker.io/kinvolk/gadget
```
`docker.io` is default ?
```suggestion
  repository: kinvolk/gadget
```

invidian reviewed (dismissed):
Just some nits, otherwise LGTM, but I'm curious why CI is failing.

iaguis commented:
I've changed the namespace to `gadget`, for some reasoning see https://github.com/kinvolk/inspektor-gadget/pull/140#issuecomment-709934072.

> I'm curious why CI is failing.

I was using `WaitForDeployment()` instead of `ẀaitForDaemonSet()` :facepalm:

iaguis commented:
> I've changed the namespace to `gadget`, for some reasoning see [kinvolk/inspektor-gadget#140 (comment)](https://github.com/kinvolk/inspektor-gadget/pull/140#issuecomment-709934072).

After thinking about it some more and discussing with others I think what makes most sense is to keep IG in the `kube-system` namespace so it works out of the box with the current `kubectl gadget` plugin. Since it will have its own PSP this should be OK. If/when upstream IG changes its namespace to `gadget` we can change it too.

I changed the PR accordingly.

invidian commented:
Odd that CI didn't get triggered.

invidian commented on docs/configuration-reference/components/inspektor-gadget.md:
```diff
@@ -0,0 +1,68 @@
+Example:
+
+| Argument           | Description                                                                                                                   | Default       | Type   | Required |
+|--------------------|-------------------------------------------------------------------------------------------------------------------------------|--:-:----------|--:-:---|--:-:-----|
```
`:-:` does not render properly on GitHub.

invidian reviewed (dismissed):
Just one broken thing, otherwise LGTM.

knrt10 commented on assets/charts/components/inspektor-gadget/values.yaml:
```diff
@@ -0,0 +1,42 @@
+# Declare variables to be passed into your templates.
+
+image:
+  repository: kinvolk/gadget
```
Unrelated but we should also push it to quay

iaguis replied:
Filed an issue upstream: https://github.com/kinvolk/inspektor-gadget/issues/141

knrt10 replied:
cool

knrt10 commented on test/components/inspektor-gadget/inspektorgadget_test.go:
```diff
@@ -0,0 +1,35 @@
+	namespace = "kube-system"
+)
+
+func TestInspektorGadgetDeployment(t *testing.T) {
```
TestInspektorGadgetDaemonSet

knrt10 reviewed (changes requested):
One small NIT, rest LGTM

knrt10 reviewed (approved):
Looks great. +1

iaguis commented:
I'll take @invidian's lgtm and @knrt10's one and merge this.
